### PR TITLE
Add support for fetch remote resources

### PR DESCRIPTION
**Pull request by ybznek**
I've add support for --fetch-remote-resources to scap-workbench

screenshot_2015-08-18_16-39-49

screenshot_2015-08-18_16-38-41

### CODE REVIEW

**Comment by ybznek** [comment]
Please, do not merge yet.
I would like to get some feedback before merging.
e.g. How to print status to "Diagnostics"?

File: include/OscapScannerBase.h [hunk 1]
```diff
@@ -62,8 +62,18 @@ class OscapScannerBase : public Scanner
 
         /// Last read rule id
         QString mLastRuleID;
-        /// If true we are in the rule ID reading phase, if false we are reading rule result
-        bool mReadingRuleID;
+        /// Last downloading file
+        QString mLastDownloadingFile;
+
+        enum Reading_state{
```
Comment (mpreisler) [inline include/OscapScannerBase.h]: ReadingState

File: include/OscapScannerBase.h [hunk 2]
```diff
@@ -62,8 +62,18 @@ class OscapScannerBase : public Scanner
...
+          READING_PREFIX,
```
Comment (mpreisler) [inline include/OscapScannerBase.h]: Prefix all the enum values with "RS_" - acronym for reading state. For example `RS_READING_PREFIX`. Check out `MessageSeverity` for an example.

File: src/OscapScannerBase.cpp [hunk 3]
```diff
@@ -270,36 +277,71 @@ bool OscapScannerBase::tryToReadStdOutChar(QProcess& process)
...
+                emit progressReport(mLastRuleID, mReadBuffer);
+                break;
+
+          case READING_DOWNLOAD_FILE_STATUS:
+             QString downloadStatus = mReadBuffer.mid(1);
+             if (downloadStatus == "ok") 
+             {
+                 emit infoMessage(QString("Downloading of \"%1\" finished: %2").arg(mLastDownloadingFile).arg(downloadStatus));
+             } else {
+                 emit warningMessage(QString("Downloading of \"%1\" finished: %2").arg(mLastDownloadingFile).arg(downloadStatus));
```
Comment (mpreisler) [inline src/OscapScannerBase.cpp]: why warning?

Comment (mpreisler) [inline src/OscapScannerBase.cpp]: Make it clearer that downloading failed

**Comment by ybznek** [comment]
Before/after : screenshot_2015-08-20_18-13-08

File: src/OscapScannerBase.cpp [hunk 4]
```diff
@@ -165,6 +167,10 @@ QStringList OscapScannerBase::buildEvaluationArgs(const QString& inputFile,
     {
         ret.append("--skip-valid");
     }
+    
+    if (mFetchRemoteResources) {
```
Comment (mpreisler) [inline src/OscapScannerBase.cpp]: code style

File: src/OscapScannerBase.cpp [hunk 5]
```diff
@@ -189,6 +195,7 @@ QStringList OscapScannerBase::buildEvaluationArgs(const QString& inputFile,
         ret.append("--tailoring-file");
         ret.append(tailoringFile);
     }
+    //
```
Comment (mpreisler) [inline src/OscapScannerBase.cpp]: ?

File: src/OscapScannerBase.cpp [hunk 6]
```diff
@@ -270,36 +277,71 @@ bool OscapScannerBase::tryToReadStdOutChar(QProcess& process)
...
-        {
-            emit progressReport(mLastRuleID, mReadBuffer);
-        }
-        else
-        {
-            emit warningMessage(QString(
-                QObject::tr("Error when parsing scan progress output from stdout of the 'oscap' process. "
-                "Newline encountered while reading rule ID, rule result and/or ':' are missing! "
-                "Read buffer is '%1'.")).arg(mReadBuffer));
+        switch(mReadingState) {
```
Comment (mpreisler) [inline src/OscapScannerBase.cpp]: code style

File: src/OscapScannerBase.cpp [hunk 7]
```diff
@@ -270,36 +277,71 @@ bool OscapScannerBase::tryToReadStdOutChar(QProcess& process)
...
+          case RS_READING_RULE_RESULT:
+                emit progressReport(mLastRuleID, mReadBuffer);
+                break;
+
+          case RS_READING_DOWNLOAD_FILE_STATUS:
+             QString downloadStatus = mReadBuffer.mid(1);
+             if (downloadStatus == "ok") 
+             {
+                 emit infoMessage(QString("Downloading of \"%1\" finished: %2").arg(mLastDownloadingFile).arg(downloadStatus));
+             } else {
```
Comment (mpreisler) [inline src/OscapScannerBase.cpp]: code style

**Comment by ybznek** [comment]
Sorry for codestyle, it is quite unusual for me and I don't realize it.

File: include/OscapScannerBase.h [hunk 8]
```diff
@@ -62,8 +62,18 @@ class OscapScannerBase : public Scanner
 
         /// Last read rule id
         QString mLastRuleID;
-        /// If true we are in the rule ID reading phase, if false we are reading rule result
-        bool mReadingRuleID;
+        /// Last downloading file
+        QString mLastDownloadingFile;
+
+        enum ReadingState{
```
Comment (mpreisler) [inline include/OscapScannerBase.h]: code style - brace on the same line, weird indent - 2 spaces

**Comment by mpreisler** [comment]
Thx, this is a great contribution, kudos!

**Pull request merged**